File: python/calculate_distances.py
"""
=========================================================================================================================
 *** calculate_distances.py ***
 Euclidean distances from every node to a reference point.
=========================================================================================================================

Description:
    Distance from every node position to one reference coordinate, used by the radius, coverage, and
    node-qualification analyses. Mirrors matlab/calculate_distances.m.

Input:
    nodePositions_m (N by 2), referencePosition_m (1 by 2)

Output:
    calculate_distances()    Distance vector in metres, length N.
=========================================================================================================================
"""
import numpy as np


def calculate_distances(nodePositions_m, referencePosition_m):
    offsets = np.asarray(nodePositions_m, float) - np.asarray(referencePosition_m, float).reshape(1, 2)
    return np.sqrt(np.sum(offsets**2, axis=1))

File: matlab/calculate_distances.m
function distances_m = calculate_distances(nodePositions_m, referencePosition_m)
%%
%% Euclidean Distances to a Reference Point:
%%
%{
Distance from every node position to one reference coordinate, used by the radius, coverage, and
node-qualification analyses.

Input:

    nodePositions_m       Node coordinates, N by 2.
    referencePosition_m   Reference coordinate, 1 by 2.

Output:

    distances_m           Distance vector in metres, N by 1.
%}

validateattributes(nodePositions_m,     {'numeric'}, {'2d', 'ncols', 2});
validateattributes(referencePosition_m, {'numeric'}, {'vector', 'numel', 2});

offsets = nodePositions_m - reshape(referencePosition_m, 1, 2);
distances_m = sqrt(sum(offsets.^2, 2));
end
